### DISCUSSION TITLE
lmp simulation of a disordered system

### DISCUSSION THREAD

**Discussion by J-You** [deepmodeling/deepmd-kit#708] [Q&A]
Hi all,

I trained a DP potential using the AIMD trajectories at liquid state. The train result seems good.
![energy](https://user-images.githubusercontent.com/80451798/120954159-1b708680-c781-11eb-9888-320db03e81b7.png)
![force](https://user-images.githubusercontent.com/80451798/120954162-1ca1b380-c781-11eb-91dc-9e8a8b3205c8.png)

Then, I run lammps by this DP potential to simulate liquid structure. The initial configuration data is disordered, but the output contains large atomic clustering.
![output](https://user-images.githubusercontent.com/80451798/120958418-4ad7c100-c78a-11eb-8fe0-42c9c798a2de.png)

The version of deepmd is 1.3.3 gpu. AIMD box contains 200 atoms. Training data is sufficient.
I have simulated other system successfully. I think it is not caused by training parameter.

Have anyone ever been in a similar situation?

Thanks for your time and attention.

**Comment by njzjz** [comment]
A possible reason is your `sel` parameter is not enough. You can check it.
You can also print out the model deviation.

**Reply by J-You** [reply]
Thanks for you response. I will try to fix it according to your suggestion.

**Reply by J-You** [reply]
Hi Jinzhe,

I trained a new model with `sel = [100, 100, 100]`. This setting is much larger than the AIMD results, around 60.
However, the similar result reproduces, that is, disorder to order in a liquid state.
The input model:
![input](https://user-images.githubusercontent.com/80451798/121450210-db521380-c9cd-11eb-8c1d-a2a9c7669d6b.png)
The output model:
![output](https://user-images.githubusercontent.com/80451798/121450231-e6a53f00-c9cd-11eb-9cbb-2f76a4d4ea7e.png)

```
{
    "_comment": " model parameters",
    "model": {
        "type_map":         ["Fe", "C", "Si"],
	"descriptor" :{
	    "type":		"se_a",
	    "sel":		[100, 100, 100],
	    "rcut_smth":	6.0,
	    "rcut":		6.2,
	    "neuron":		[25, 50, 100],
	    "resnet_dt":	false,
	    "axis_neuron":	12,
	    "seed":		165498106467,
	    "_comment":		" that's all"
	},
	"fitting_net" : {
	    "neuron":		[240, 240, 240],
	    "resnet_dt":	true,
	    "seed":		1464987461649,
	    "_comment":		" that's all"
	},
	"_comment":	" that's all"
    },
    "learning_rate" :{
	"type":		"exp",
	"start_lr":	0.001,
	"decay_steps":	5000,
        "decay_rate":   0.93,
	"_comment":	"that's all"
    },
    "loss" :{
	"start_pref_e":	0.01,
	"limit_pref_e":	1,
	"start_pref_f": 1000,
	"limit_pref_f":	1,
	"start_pref_v":	0,
	"limit_pref_v":	0,
	"_comment":	" that's all"
    },

    "_comment": " traing controls",
    "training" : {
	"systems":	["../data/"],
	"set_prefix":	"set",    
	"stop_batch":	2000000,
	"batch_size":	1,
	"seed":		4561654871,
	"_comment": " display and restart",
	"_comment": " frequencies counted in batch",
	"disp_file":	"lcurve.out",
	"disp_freq":	100,
	"numb_test":	10,
	"save_freq":	1000,
	"save_ckpt":	"model.ckpt",
	"load_ckpt":	"model.ckpt",
	"disp_training":true,
	"time_training":true,
	"profiling":	false,
	"profiling_file":"timeline.json",
	"_comment":	"that's all"
    },

    "_comment":		"that's all"
}
```
Could you please take a look of my training parameters?
I have trained the binary system successfully. Could it be caused by the preparation data?

**Reply by njzjz** [reply]
Did you also check MD trajectories driven by DP?

**Reply by J-You** [reply]
The trajectories are also unreasonable.
Maybe I need more precise training data.
Thanks for your response.

**Reply by rustin-domingos** [reply]
Not sure if this was resolved, but I had a similar issue when using a model to run MD, I had incorrect scaling in my input file. I believe I had direct or reduced coordinates rather than cartesian coordinates. Once I corrected this I no longer saw this behavior.

**Reply by J-You** [reply]
Thanks for your response. This problem has not been fixed. Could you please tell me how do you generate training set? I use `dpdata` converting `OUTCAR` to `.npy` directly.
```
from dpdata import LabeledSystem
from glob import glob

xml=['OUTCAR_perturb1', 'OUTCAR_perturb2', 'OUTCAR_perturb3']
total_system=LabeledSystem()
for f in xml:
    ls=LabeledSystem(f, fmt='vasp/outcar')
    total_system.append(ls)
total_system.shuffle()
total_system.to_deepmd_npy('../data/train_set/crystal', set_size=5000)
```

**Reply by rustin-domingos** [reply]
Yes, I did the same, however for the initial configuration for MD verify the lammps input file is not in reduced coordinates. 

I had initially written my own MD code and tried calling dp.eval(…) with a POSCAR in DIRECT coordinates as my initial configuration, I saw behavior just like yours. When I scaled it up to cartesian it worked correctly. 

I was confused because i had trained directly from OUTCARs as you have done and assumed dp.eval would work on reduced coordinates. Hopefully this helps you!

**Reply by J-You** [reply]
I see. It is a trick. Thanks!

**Reply by J-You** [reply]
Hi,
I tried to use Cartesian coordinates configuration as the input for LAMMPS simulation.
![微信图片_20210826104438](https://user-images.githubusercontent.com/80451798/130891845-a532bcad-8b5a-477b-89c0-e5803b5c12b0.png)
However, this problem was not solved in my ternary system.

As for the simulation of a multi-element system driven by deep potential, the quality of my training data may need improvement.


I also want to consult, if an element contains too few atoms in the AIMD simulation, is the accuracy of deep potential going to be reduced?

**Reply by 343333333** [reply]
i used to train 64H2O model in dpmd , i meet the same problem like you . after i change my sel  and it seems work well. A smaller sampling interval may also be effective

**Reply by J-You** [reply]
Thanks for your response. Do you mean that the `sel' is too small in my situation?
Could you please tell me more about how to change sampling interval, does it refers to batch size?

**Reply by 343333333** [reply]
> Thanks for your response. Do you mean that the `sel' is too small in my situation?
> Could you please tell me more about how to change sampling interval, does it refers to batch size?

-its a odd thing to choose a good sel , i used sel of 96 which have bad performance on lammps , but 98  has a good performance . So , you should test a lot
-I mean the sampling interval t in lammps , water system performance better(less chance to bankrupt ) with the interval of 0.5fs than 1fs, because of the hydrogen.

**Reply by J-You** [reply]
Really appreciate for your suggestion. It does need a lot of test in my own training model.

**Reply by zhuhaotian519** [reply]
Hi, I had a similar problem with the DP model causing the MD trajectory to be messed up, how did you fix it later? Thank you!
